source/overview/faq.rst

```diff
@@ -265,29 +269,157 @@ Working for Mattermost
+
+We expect to keep plugins without an enterprise component under our open source license. No licensing changes are planned to non-plugin repositories, such as mattermost-server or mattermost-webapp.
+
+Will the repositories be public?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Yes, existing repositories will stay public. We are now also able to make public several enterprise-only plugins previously developed in private.
+
+Can I still contribute?
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Yes, we continue to welcome all contributions. Mattermost may select some contributions as enterprise features and license them under the Mattermost Source Available License. As with all Mattermost repositories, you will still need to sign the `Mattermost CLA <https://mattermost.org/mattermost-contributor-agreement/>`_. We will not accept contributions without signing the Mattermost CLA.
+
+Do I need to re-sign the `Mattermost CLA <https://mattermost.org/mattermost-contributor-agreement/>`_?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+No, if you have already signed the `Mattermost CLA <https://mattermost.org/mattermost-contributor-agreement/>`_, you do not need to sign it again.
+
+Can I compile your plugins by myself?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Yes. If you have a Mattermost Enterprise E20 license, you are free to compile and use a plugin under the Mattermost Source Available License. Furthermore, if you are developing against or testing with such a plugin, you are free to compile and test a plugin even without a Mattermost Enterprise E20 license. Without an Enterprise E20 license, source available plugins may have reduced functionality or refuse to start altogether. Request a `trial license <https://mattermost.com/trial/>`_ if your testing requires access to enterprise functionality.
+
+Several of our customers value complete access to our source code and compile our plugins from scratch before deploying to their production servers. By adopting the Mattermost Source Available License, we can develop enterprise-only features in public without impacting this workflow.
+
+Will you distribute open-source plugin binaries without any licensing restrictions?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+At this time, we have no plans to distribute more than one version of each of our plugins. Without a Mattermost Enterprise E20 License, plugins may have reduced functionality or refuse to start altogether.
+
+Can I continue to use the existing open-source repositories without restriction?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Yes, the Mattermost Source Available License will only apply from the date it is added and to the versions in which it is included.
+
+Can I publish my own plugin under a license of my choosing?
```

Comment:
This confused me now that I'm reading this question again. At first I thought this meant I can publish my own version of the source available plugin under a license of my choosing. However, I believe this means a different plugin I've built.

I wonder how much value this question adds, @lieut-data?

Reply:
I've suggested some rewording.